doc/rtd/explanation/hardening.rst

```diff
@@ -0,0 +1,80 @@
+Security Hardening
+******************
+
+Cloud-init's use case is automating cloud instance initialization, with support
+across distributions and platforms. There is a myriad of ways to harden this
+space.
+
+Follow the security hardening guidelines provided by the OSes and cloud
+platforms that your cloud-init configuration is targeting.
+
+Many cloud platforms provide SSH public keys in metadata which setup the
+default user with the appropriate configured means of access using
+SSH public/private key pairs.
+
+
+Updated packages
+================
+
+To ensure the available security fixes are applied to you VMs images upon
+launch, it is recommended to update the packages
+
+.. note::
+
+  Ubuntu cloud images are configured by default to enable unattended-upgrades,
+  thus this is resolved this issue when the update gets triggered. One can
+  still apply this recommendation to cloud that gap and update the packages
+  on first boot.
+
+.. code-block:: yaml
+
+  #cloud-config
+  package_update: true
+  package_upgrade: true
+
+
+Ubuntu's security guidelines: https://documentation.ubuntu.com/server/explanation/security/security_suggestions/#keep-your-system-up-to-date
```

Comment:
Let's drop this as an embeded link inline above the section and provide the URL in the footer in RST
```suggestion
```